### PR TITLE
Remove old ASB variable

### PR DESCRIPTION
**Pull request by geoand**
Fixes: #103

### CODE REVIEW

**Comment by geoand** [comment]
I am rerunning the playbooks now to make sure everything works as expected

**Comment by geoand** [comment]
Using openshift-ansible version: `openshift-ansible-3.10.46-1` and the changes in the PR, ASB was installed correctly:

```
﻿Using config file mounted to /etc/ansible-service-broker/config.yaml
============================================================
==           Creating Ansible Service Broker...           ==
============================================================
time="2018-09-11T06:14:53Z" level=info msg="Ansible Service Broker Version: 1.3.15"
time="2018-09-11T06:14:53Z" level=info msg="Initializing clients..."
time="2018-09-11T06:14:53Z" level=info msg="OpenShift version: v3.10.0+e3465d0-44"
time="2018-09-11T06:14:53Z" level=info msg="Kubernetes version: v1.10.0+b81c8f8"
time="2018-09-11T06:14:53Z" level=info msg="Empty AuthType. Assuming credentials are defined in the config... "
time="2018-09-11T06:14:53Z" level=info msg="== REGISTRY CX == "
time="2018-09-11T06:14:53Z" level=info msg="Name: dh"
time="2018-09-11T06:14:53Z" level=info msg="Type: dockerhub"
time="2018-09-11T06:14:53Z" level=info msg="Url: "
time="2018-09-11T06:14:53Z" level=info msg="Empty AuthType. Assuming credentials are defined in the config... "
time="2018-09-11T06:14:53Z" level=info msg="== REGISTRY CX == "
time="2018-09-11T06:14:53Z" level=info msg="Name: localregistry"
time="2018-09-11T06:14:53Z" level=info msg="Type: local_openshift"
time="2018-09-11T06:14:53Z" level=info msg="Url: "
============================================================
==           Starting Ansible Service Broker...           ==
============================================================
time="2018-09-11T06:14:53Z" level=info msg="Initiating Recovery Process"
time="2018-09-11T06:14:53Z" level=info msg="Recovery complete"
time="2018-09-11T06:14:53Z" level=info msg="recover called"
time="2018-09-11T06:14:53Z" level=info msg="Broker configured to bootstrap on startup"
time="2018-09-11T06:14:53Z" level=info msg="Attempting bootstrap..."
time="2018-09-11T06:14:53Z" level=info msg="AnsibleBroker::Bootstrap"
time="2018-09-11T06:14:53Z" level=info msg="0 specs deleted"
time="2018-09-11T06:14:57Z" level=info msg="Bundles filtered by white/blacklist filter:\n\t-> ansibleplaybookbundle/kubevirt-ansible\n\t-> ansibleplaybookbundle/hello-world-apb\n\t-> ansibleplaybookbundle/rocketchat-apb\n\t-> ansibleplaybookbundle/rds-postgres-apb\n\t-> ansibleplaybookbundle/etherpad-apb\n\t-> ansibleplaybookbundle/wordpress-ha-apb\n\t-> ansibleplaybookbundle/hello-world-db-apb\n\t-> ansibleplaybookbundle/hastebin-apb\n\t-> ansibleplaybookbundle/pyzip-demo-apb\n\t-> ansibleplaybookbundle/jenkins-apb\n\t-> ansibleplaybookbundle/pyzip-demo-db-apb\n\t-> ansibleplaybookbundle/thelounge-apb\n\t-> ansibleplaybookbundle/manageiq-apb\n\t-> ansibleplaybookbundle/nginx-apb\n\t-> ansibleplaybookbundle/mediawiki-apb\n\t-> ansibleplaybookbundle/postgresql-apb\n\t-> ansibleplaybookbundle/mysql-apb\n\t-> ansibleplaybookbundle/mariadb-apb\n\t-> ansibleplaybookbundle/s2i-apb\n\t-> ansibleplaybookbundle/es-apb\n\t-> ansibleplaybookbundle/kibana-apb\n\t-> ansibleplaybookbundle/prometheus-apb\n\t-> ansibleplaybookbundle/awx-apb\n\t-> ansibleplaybookbundle/photo-album-demo-api-apb\n\t-> ansibleplaybookbundle/photo-album-demo-app-apb\n\t-> ansibleplaybookbundle/photo-album-demo-ext-api-apb\n\t-> ansibleplaybookbundle/iscsi-demo-target-apb\n\t-> ansibleplaybookbundle/mongodb-apb\n\t-> ansibleplaybookbundle/keycloak-apb\n\t-> ansibleplaybookbundle/virtualmachines-apb\n\t-> ansibleplaybookbundle/proxy-config-apb\n\t-> ansibleplaybookbundle/mssql-remote-apb\n\t-> ansibleplaybookbundle/dynamic-apb\n\t-> ansibleplaybookbundle/mssql-apb\n\t-> ansibleplaybookbundle/eclipse-che-apb\n\t-> ansibleplaybookbundle/manageiq-runner-apb\n\t-> ansibleplaybookbundle/import-vm-apb\n\t-> ansibleplaybookbundle/kubevirt-apb\n\t-> ansibleplaybookbundle/v2v-apb\n\t-> ansibleplaybookbundle/vnc-desktop-apb\n\t-> ansibleplaybookbundle/galera-apb\n\t-> ansibleplaybookbundle/openshift-logging-apb\n\t-> ansibleplaybookbundle/rhpam-apb\n\t-> ansibleplaybookbundle/tiller-apb\n\t-> ansibleplaybookbundle/standalone-cinder-apb\n\t-> ansibleplaybookbundle/gluster-s3object-apb\n\t-> ansibleplaybookbundle/origin-ansible-service-broker\n\t-> ansibleplaybookbundle/hello-world\n\t-> ansibleplaybookbundle/mediawiki123\n\t-> ansibleplaybookbundle/apb-base\n\t-> ansibleplaybookbundle/ansible-service-broker\n\t-> ansibleplaybookbundle/apb-tools\n\t-> ansibleplaybookbundle/manageiq-apb-runner\n\t-> ansibleplaybookbundle/py-zip-demo\n\t-> ansibleplaybookbundle/mediawiki\n\t-> ansibleplaybookbundle/asb-installer\n\t-> ansibleplaybookbundle/apb-assets-base\n\t-> ansibleplaybookbundle/photo-album-demo-app\n\t-> ansibleplaybookbundle/helm-bundle-base\n\t-> ansibleplaybookbundle/origin\n\t-> ansibleplaybookbundle/photo-album-demo-api\n\t-> ansibleplaybookbundle/deploy-broker\n\t-> ansibleplaybookbundle/vnc-client\n\t-> ansibleplaybookbundle/origin-service-catalog\n\t-> ansibleplaybookbundle/vnc-desktop\n\t-> ansibleplaybookbundle/origin-deployer\n\t-> ansibleplaybookbundle/origin-docker-registry\n\t-> ansibleplaybookbundle/origin-haproxy-router\n\t-> ansibleplaybookbundle/origin-pod\n\t-> ansibleplaybookbundle/origin-sti-builder\n\t-> ansibleplaybookbundle/origin-recycler\n"
time="2018-09-11T06:14:57Z" level=info msg="Validating specs..."
time="2018-09-11T06:14:57Z" level=info msg="All specs passed validation!"
time="2018-09-11T06:14:57Z" level=info msg="Bundles filtered by white/blacklist filter:\n\t-> openshift/dotnet\n\t-> openshift/dotnet-runtime\n\t-> openshift/httpd\n\t-> openshift/jenkins\n\t-> openshift/mariadb\n\t-> openshift/mongodb\n\t-> openshift/mysql\n\t-> openshift/nginx\n\t-> openshift/nodejs\n\t-> openshift/perl\n\t-> openshift/php\n\t-> openshift/postgresql\n\t-> openshift/python\n\t-> openshift/redis\n\t-> openshift/ruby\n\t-> openshift/wildfly\n"
time="2018-09-11T06:14:57Z" level=info msg="Validating specs..."
time="2018-09-11T06:14:57Z" level=info msg="All specs passed validation!"
time="2018-09-11T06:14:57Z" level=info msg="Broker successfully bootstrapped on startup"
time="2018-09-11T06:14:57Z" level=info msg="Listening on https://[::]:1338"
time="2018-09-11T06:14:57Z" level=info msg="Ansible Service Broker Starting"
```

File: ansible/playbook/roles/generate_inventory/templates/cloud.inventory.j2 [hunk 1]
```diff
@@ -39,10 +39,7 @@ openshift_master_htpasswd_users={'admin': '$apr1$DloeoaY3$nqbN9fQBkyXgbj58buqEM.
 
 openshift_node_groups=[{'name': 'node-config-all-in-one', 'labels': ['node-role.kubernetes.io/master=true', 'node-role.kubernetes.io/infra=true', 'node-role.kubernetes.io/compute=true'], 'edits': [{ 'key': 'kubeletArguments.pods-per-core','value': ['20']}]}]
 
-# Used to control the image version of origin-ansible-service-broker
-# see: https://github.com/openshift/openshift-ansible/blob/release-3.10/roles/ansible_service_broker/defaults/main.yml
-# Note: the variable seems to not be used currently, but that is most likely a bug
-ansible_service_broker_image_tag=1.2
+ansible_service_broker_registry_tag=release-1.2
```
[the remark this answers is not in the document]
Comment (cmoulliard) [reply ansible/playbook/roles/generate_inventory/templates/cloud.inventory.j2]: If I look to the role definition, I don't see that we can change the value defined here : https://goo.gl/NRW2qE

[the remark this answers is not in the document]
Comment (geoand) [reply ansible/playbook/roles/generate_inventory/templates/cloud.inventory.j2]: Do we know that `1.2` is for Openshift `3.10` and `1.3` is for Openshift `3.11`?

[the remark this answers is not in the document]
Comment (geoand) [reply ansible/playbook/roles/generate_inventory/templates/cloud.inventory.j2]: According to [this](https://github.com/openshift/ansible-service-broker#release-dates), that is indeed the case

[the remark this answers is not in the document]
Comment (geoand) [reply ansible/playbook/roles/generate_inventory/templates/cloud.inventory.j2]: I think we should we to see what the ASB team will reply to your questions/issues

[the remark this answers is not in the document]
Comment (cmoulliard) [reply ansible/playbook/roles/generate_inventory/templates/cloud.inventory.j2]: No. Jason Montlean published this info on the ASB mailling list

```
Someone recently filed a bug for the v3.10/latest version mismatch. We have some ideas how to handle it better and are working on implementing it for 3.11 (and possibly 3.10).

For now it's pretty easy to modify the dc image to get the release you want.

oc patch dc -n openshift-automation-service-broker openshift-automation-service-broker --patch='{"spec":{"template":{"spec":{"containers":[{"name": "penshift-automation-service-broker", "image":"ansibleplaybookbundle/origin-ansible-service-broker:release-1.2"}]}}}}'

release-1.0 corresponds to 3.7
release-1.1 3.9
release-1.2 3.10
latest what will become release-1.3 3.11

I'm not sure anything earlier than release-1.2 will work with the resources created by the APB so I'd stick to release-1.2 and latest if you can for now.
```

[the remark this answers is not in the document]
Comment (geoand) [reply ansible/playbook/roles/generate_inventory/templates/cloud.inventory.j2]: Sounds like a mess

[the remark this answers is not in the document]
Comment (cmoulliard) [reply ansible/playbook/roles/generate_inventory/templates/cloud.inventory.j2]: I changed the docker image version to `docker.io/ansibleplaybookbundle/origin-ansible-service-broker:release-1.2` but I don't get/see any `*-apb` services from the catalog ....

[the remark this answers is not in the document]
Comment (geoand) [reply ansible/playbook/roles/generate_inventory/templates/cloud.inventory.j2]: Let me try that as well

[the remark this answers is not in the document]
Comment (cmoulliard) [reply ansible/playbook/roles/generate_inventory/templates/cloud.inventory.j2]: As `ansible_service_broker_registry_tag` is used as a parameter of the ConfigMap loaded by the ASB's pod, then we can't use it to define the version of ASB or OAB broker docker image to be pulled.  I propose that we remove for the moment ASB parameters from the template used to install origin 3.10. Make sense @geoand ?

[the remark this answers is not in the document]
Comment (geoand) [reply ansible/playbook/roles/generate_inventory/templates/cloud.inventory.j2]: Yes, I agree

[the remark this answers is not in the document]
Comment (geoand) [reply ansible/playbook/roles/generate_inventory/templates/cloud.inventory.j2]: PR updated

**Pull request merged**